pkgs/servers/vpp/default.nix

```diff
@@ -0,0 +1,152 @@
+{ lib
+, llvmPackages_14
+, fetchgit
+, fetchurl
+, callPackage
+, cmake
+, dpdk
+, intel-ipsec-mb
+, xdp-tools
+, libbpf
+, libmnl
+, elfutils
+, zlib
+, rdma-core
+, python310
+, python310Packages
+, openssl
+, libuuid
+, subunit
+, pkg-config
+, libpcap
+, jansson
+, libnl
+, libdaq
+, srtp
+, mbedtls_2
+, check
+, nixosTests
+}:
+
+let
+  stdenv = llvmPackages_14.stdenv;
+
+  xdp-tools' = xdp-tools.overrideAttrs (old: rec {
+    postInstall = ''
+      # Drop unfortunate references to glibc.dev/include at least from $lib
+      nuke-refs "$lib"/lib/bpf/*.o
+    '';
+  });
+
+  intel-ipsec-mb' = intel-ipsec-mb.overrideAttrs (old: rec {
+    makeFlags = old.makeFlags ++ [
+      "SHARED=n"
+    ];
+  });
+
+  dpdk' = dpdk.overrideAttrs (old: rec {
+    version = "22.07";
+    src = fetchurl {
+      url = "https://fast.dpdk.org/rel/dpdk-${version}.tar.xz";
+      sha256 = "sha256-n2Tf3gdf21cIy2Leg4uP+4kVdf7R4dKusma6yj38m+o";
+    };
+  });
+
+  rdma-core' = rdma-core.overrideAttrs (old: rec {
+    cmakeFlags = old.cmakeFlags ++ [
+      "-DENABLE_STATIC=1"
+    ];
+  });
+
+  srtp' = srtp.overrideAttrs (old: rec {
+    mesonFlags = old.mesonFlags ++ [
+      "-Ddefault_library=static"
+    ];
+  });
+in
+stdenv.mkDerivation rec {
+  pname = "vpp";
+  version = "23.06";
+  src = fetchgit {
+    url = "https://gerrit.fd.io/r/vpp";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-9dn/rpjouwjFUFoQYd8Go1rV4ThZ8gh/egIuXfdPys0=";
+  };
+
+  patches = [
+    ./0001-fix-nsh-plugin-loading.patch
+    ./0002-explicity-include-std-array.patch
+  ];
+
+  sourceRoot = "vpp/src";
+
+  quicly = callPackage ./quicly { };
```

Comment:
Since quicly also provides a cli application, and could theoretically be used by other software, please add it to all-packages and take it as input of vpp like any other package.